talksapp/main_test.go

```diff
@@ -0,0 +1,293 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd.
+
+package talksapp
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/golang/gddo/gosrc"
+
+	"appengine"
+	"appengine/aetest"
+	"appengine/memcache"
+)
+
+const importPath = "github.com/user/repo/path/to/presentation.slide"
+
+func TestHome(t *testing.T) {
+	i, err := aetest.NewInstance(nil)
```

Comment:
Define an extra function to factor out all the instance and request creation:

```go
func do(t *testing.T, method, path string, f func(*http.Request)) {
        i, err := aetest.NewInstance(nil)
        if err != nil {
                t.Fatal(err)
        }   
        defer i.Close()

        r, err := i.NewRequest(method, path, nil)
        if err != nil {
                t.Fatal(err)
        }   

        f(r)
}
```

Then your code can be as simple as:

```go
func TestHome(t *testing.T) {
        do(t, "GET", "/", func(r *http.Request) {
                w := httptest.NewRecorder()
                handlerFunc(serveRoot).ServeHTTP(w, r)
                if w.Code != http.StatusOK {
                        t.Fatalf("expected status: %d, got: %d", http.StatusOK, w.Code)
                }   

                if !strings.Contains(w.Body.String(), "go-talks.appspot.org") {
                        t.Fatal("expected response to contain: go-talks.appspot.org")
                }   
        })  
}
```